### PR TITLE
Avoid setting `hasAnyCodeStyleOption` for analyzers with no `CodeStyleOption2<T>`

### LINKED ISSUE #72162: Remove unnecessary imports analyzer is significantly slowing builds (11% CPU time)
image

@mavasani How can this be disabled? I already have `dotnet_diagnostic.IDE0005.severity = none` in `.globalconfig` file.

### LINKED ISSUE #72876: AbstractFileHeaderDiagnosticAnalyzer uses `CustomSeverityConfigurable` but does not provide a way to set the custom severity
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Analyzer-ignores-settings/10625971)._

---
[severity:I'm unable to use this version] [regression] [worked-in:`17.7.5`]
Analyzer ignores settings
the file (.editorconfig) contains the header settings

```
# File header preferences
dotnet_diagnostic.IDE0073.severity = warning
file_header_template = <copyright file="{fileName}" company="MyCompany">\nCopyright (c) Company. All rights reserved.\n</copyright>
```
and .csproj contains
```
  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
    <WarningsAsErrors />
  </PropertyGroup>
```
before updating to the current version, during assembly, if I did not specify the header in the file - I received an error, now this setting is ignored

sample of file header
```
// <copyright file="NewStateTests.cs" company="MyCompany">
// Copyright (c) Company. All rights reserved.
// </copyright>
```

---
#### Feedback Bot on 3/27/2024, 11:48 PM: 

(private comment, text removed)
#### Wenwen Fan [MSFT] on 3/28/2024, 04:09 AM: 

(private comment, text removed)
#### Wenwen Fan [MSFT] on 3/28/2024, 04:10 AM: 

(private comment, text removed)
#### Александр Платоненков on 3/30/2024, 05:02 AM: 

(private comment, text removed)
#### Александр Платоненков on 4/4/2024, 10:56 AM: 

(private comment, text removed)
#### Wenwen Fan [MSFT] on 4/3/2024, 01:40 AM: 

(private comment, text removed)
#### Sam Harwell [MSFT] on 4/4/2024, 08:05 AM: 

(private comment, text removed)
#### Sam Harwell on 4/4/2024, 11:38 AM: 

<p>Hi <a target='_blank' href="https://developercommunity.visualstudio.com/user/1315100/profile">Александр Платоненков</a>,</p>
<p>You can ignore the previous request for more information. I found the source of the problem.</p>
<p>Thank you for the report and quick responses!</p>

---
### Original Solutions
(no solutions)

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #72162
Fixes #72876

### CODE REVIEW

File: src/Analyzers/Core/Analyzers/FileHeaders/AbstractFileHeaderDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ internal abstract class AbstractFileHeaderDiagnosticAnalyzer : AbstractBuiltInCo
     private static readonly DiagnosticDescriptor s_missingHeaderDescriptor = CreateDescriptorForFileHeader(s_missingHeaderTitle, s_missingHeaderMessage);
 
     private static DiagnosticDescriptor CreateDescriptorForFileHeader(LocalizableString title, LocalizableString message)
-        => CreateDescriptorWithId(IDEDiagnosticIds.FileHeaderMismatch, EnforceOnBuildValues.FileHeaderMismatch, hasAnyCodeStyleOption: true, title, message);
+        => CreateDescriptorWithId(IDEDiagnosticIds.FileHeaderMismatch, EnforceOnBuildValues.FileHeaderMismatch, hasAnyCodeStyleOption: false, title, message);
```
Comment (ToddGrun) [inline src/Analyzers/Core/Analyzers/FileHeaders/AbstractFileHeaderDiagnosticAnalyzer.cs:24]: >false [](http://example.com/codeflow?start=135&length=5)

I tried this earlier at your suggestion, and I believe it caused me to hit asserts. I assume you don't see those? If not, I can try building this branch locally and see if I can still reproduce.

Comment (ToddGrun) [reply src/Analyzers/Core/Analyzers/FileHeaders/AbstractFileHeaderDiagnosticAnalyzer.cs:24]: If I'm remembering, the assert was that options were specifies but that this flag was false.

Comment (sharwell) [reply src/Analyzers/Core/Analyzers/FileHeaders/AbstractFileHeaderDiagnosticAnalyzer.cs:24]: > I assume you don't see those?

I haven't tried running this. If CI shows asserts I can review.

Comment (sharwell) [reply src/Analyzers/Core/Analyzers/FileHeaders/AbstractFileHeaderDiagnosticAnalyzer.cs:24]: Found one such issue, now running tests again.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**